src/shared/calendarspec.c

```diff
@@ -620,6 +628,39 @@ static int calendarspec_from_time_t(CalendarSpec *c, time_t time) {
         return 0;
 }
 
+static bool is_single_value_component(CalendarComponent *cc) {
+        assert(cc);
+        return cc->stop == -1 && cc->repeat == 0 && cc->next == NULL;
```

Comment:
Please use implicit cast to boolean for pointer:
`cc->next == NULL` should be `!cc->next`